### PR TITLE
Add uv_timer_t wrappers to libuv layer

### PR DESCRIPTION
**Pull request by cesarbs**
cc @halter73 @mikeharder @davidfowl

### CODE REVIEW

**Comment by benaadams** [comment]
`System.Threading.Timer`? 

Has the advantage if they don't perform well you know where they work ;-)

File: src/Microsoft.AspNetCore.Server.Kestrel/Internal/Networking/UvTimerHandle.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+using System;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Infrastructure;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Internal.Networking
+{
+    public class UvTimerHandle : UvHandle
+    {
+        private readonly static Libuv.uv_timer_cb _uv_timer_cb = (handle) => UvTimerCb(handle);
```
Comment (txdv) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/Networking/UvTimerHandle.cs]: Why not _uv_timer_cb = UvTimerCb? It seems like the signatures match.

**Comment by txdv** [comment]
What are the timers going to be used for?

**Comment by davidfowl** [comment]
Timeouts

**Comment by cesarbs** [comment]
@benaadams We've decided to go with uv timers to keep them in the event loop. With framework timers we'd have to post a lot (with the current plan for implementing timeouts).

**Comment by halter73** [comment]
Looks reasonable. I'll hold off on reviewing until I see how it's used.

**Comment by cesarbs** [comment]
Closing this since #1066 has the same changes.

**Pull request closed without merging**